polaris-react/src/components/CustomProperties/CustomProperties.scss

```diff
@@ -0,0 +1,7 @@
+/* stylelint-disable scss/at-import-partial-extension-blacklist */
+
+/**
+ * Import the scss file from polaris-tokens
+ * so that the content is appended to the main polaris-react css file
+ */
+@import '../../../../node_modules/@shopify/polaris-tokens/dist/scss/styles.scss';
```

Comment:
This should get cleaned up in the future and import from the built package. Could this break for a consumer that is using the scss and building it?

Reply:
> Could this break for a consumer that is using the scss and building it?

Did some digging and with the recent removal of the public sass API we no longer expose any raw SCSS files for consumers to build.

> This should get cleaned up in the future and import from the built package.

We tried this originally and were having issues configuring `node-sass` to resolve `styles.scss` from `@shopify/polaris-tokens`. Agree this is something we should circle back to 👍 Would love this too look more like:
```scss
@import '@shopify/polaris-tokens/scss/styles.scss'
```

Reply:
Yes. I would just adjust it slightly too... so that it's clear this is not a thing we do for other tokens.
```scss
@import '@shopify/polaris-tokens/scss/breakpoints.scss'
```

Reply:
As Aaron says, scss files in polaris-react are compiled at build time and we ship css files. There are no scss file in the published `@shopify/polaris` package anymore. Thus having this funky relative import won't break anything - it just looks ugly AF.

https://github.com/Shopify/polaris/blob/main/polaris-react/config/rollup/plugin-styles.js#L155 is where we teach sass about load paths - we should be able to teach it to look in node_modules , so you can do `@import '@shopify/polaris-tokens'`. See https://sass-lang.com/documentation/at-rules/import#load-paths


---
As a side note - you'll be able to cmd+click on this relative path in vscode to go to the tokens file. I believe you'll loose that functionality if you do a `@import '@shopify/polaris-tokens'`. Is that DX tradeoff important?

Reply:
Yes! My misunderstanding I was thinking of the public API that no longer exists 🤙

Reply:
@BPScott Thanks for pointing me in the right direction. I'll give that a shot!

Reply:
@alex-page 

> Yes. I would just adjust it slightly... so that it's clear this is not a thing we do for other tokens.
> ```@import '@shopify/polaris-tokens/scss/breakpoints.scss'```

Do you mean `@import '@shopify/polaris-tokens/scss/custom-properties.scss`? The `styles.scss` file contains all our Polaris custom properties (no breakpoints).

Reply:
I would rename that file from `style.scss` to `breakpoints.scss` as this is an edge case for breakpoints today. This isn't a pattern we want to scale for multiple tokens.

Reply:
Oh my bad. I was thinking we would import them with css not scss. That makes sense though.